mrmustard/lab_dev/circuit_components.py

```diff
@@ -159,6 +160,37 @@ def light_copy(self) -> CircuitComponent:
         instance.__dict__["_wires"] = self.wires.copy()
         return instance
 
+    def to_fock(self, shape: Optional[Union[int, Iterable[int]]] = None) -> CircuitComponent:
+        r"""
+        Returns a circuit component with the same attributes as this component, but
+        with ``Fock`` representation.
+
+        Uses the :meth:`mrmustard.physics.converters.to_fock` method to convert the internal
+        representation.
+
+        .. code-block::
+
+            >>> from mrmustard.physics.converters import to_fock
+            >>> from mrmustard.lab_dev import Dgate
+
+            >>> d = Dgate([1], x=0.1, y=0.1)
+            >>> d_fock = d.to_fock(shape=3)
+
+            >>> assert d_fock.name == d.name
+            >>> assert d_fock.wires == d.wires
+            >>> assert d_fock.representation == to_fock(d.representation, shape=3)
+
+        Args:
+            shape: The shape of the returned representation. If ``shape``is given as
+                an ``int``, it is broadcasted to all the dimensions. If ``None``, it
+                defaults to the value of ``AUTOCUTOFF_MAX_CUTOFF`` in the settings.
+        """
+        return CircuitComponent.from_attributes(
+            self.name,
+            to_fock(self.representation, shape=shape),
```

Comment:
I think it can be confusing to use the same name as the function we're defining. Let's think of alternative naming.